### PR TITLE
chore: Add better error handling for session creation responses

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: test/unit/helper/test_helper.py [hunk 1]
```diff
@@ -50,29 +50,27 @@ def android_w3c_driver() -> 'WebDriver':
 
     response_body_json = json.dumps(
         {
-            'value': {
```
[the remark this answers is not in the document]
Comment (KazuCocoa) [reply test/unit/helper/test_helper.py:53]: Old  protocol has `session id` on the top, btw.
https://www.selenium.dev/documentation/legacy/json_wire_protocol/

[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply test/unit/helper/test_helper.py:53]: the session response body must include sessionId (and) capabilities. `value` is more about in-session responses

[the remark this answers is not in the document]
Comment (KazuCocoa) [reply test/unit/helper/test_helper.py:53]: mm, but this CI had `appium@next` and it passed. Maybe I missed something in this change.
Let me see later again.

[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply test/unit/helper/test_helper.py:53]: https://source.chromium.org/chromium/chromium/src/+/main:chrome/test/chromedriver/session_commands.cc;l=336?q=%22sessionId%22%20capabilities&ss=chromium

[the remark this answers is not in the document]
Comment (KazuCocoa) [reply test/unit/helper/test_helper.py:53]: thanks for the link. yea, it seems the body... below is chrome 102.0.1245.33 one: 
```
$ curl -i -X POST http://localhost:9515/session -d '{"capabilities": {"browserNAme": "chrome"}}'
HTTP/1.1 200 OK
Content-Length:769
Content-Type:application/json; charset=utf-8
cache-control:no-cache

{"value":{"capabilities":{"acceptInsecureCerts":false,"browserName":"msedge","browserVersion":"102.0.1245.41","ms:edgeOptions":{"debuggerAddress":"localhost:54756"},"msedge":{"msedgedriverVersion":"102.0.1245.33 (41285f1c5672f618820f679d5efba8bafa4828de)","userDataDir":"/var/folders/34/2222sh8n27d6rcp7jqlkw8km0000gn/T/.com.microsoft.edgemac.Un8VWC"},"networkConnectionEnabled":false,"pageLoadStrategy":"normal","platformName":"mac os x","proxy":{},"setWindowRect":true,"strictFileInteractability":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify","webauthn:extension:credBlob":true,"webauthn:extension:largeBlob":true,"webauthn:virtualAuthenticators":true},"sessionId":"91f027e4062c2ce567a86ead3cd65e6a"}}
```
gecko and safari also followed the format, so I also asked Titus about this...
Potentially this browser side implementation is on the middle of w3c spec support.

[the remark this answers is not in the document]
Comment (KazuCocoa) [reply test/unit/helper/test_helper.py:53]: oh, sorry i forgot paste the correct one. (actually my previous comment was wrong format)


```
$ curl -i -X POST http://localhost:9515/session -d '{"capabilities": {"alwaysMatch": {"browserName":"chrome"}}, "firstMatch": [{}] }'
HTTP/1.1 200 OK
Content-Length:800
Content-Type:application/json; charset=utf-8
cache-control:no-cache

{"value":{"capabilities":{"acceptInsecureCerts":false,"browserName":"chrome","browserVersion":"102.0.5005.115","chrome":{"chromedriverVersion":"102.0.5005.61 (0e59bcc00cc4985ce39ad31c150065f159d95ad3-refs/branch-heads/5005@{#819})","userDataDir":"/var/folders/34/2222sh8n27d6rcp7jqlkw8km0000gn/T/.com.google.Chrome.gCsuyX"},"goog:chromeOptions":{"debuggerAddress":"localhost:55457"},"networkConnectionEnabled":false,"pageLoadStrategy":"normal","platformName":"mac os x","proxy":{},"setWindowRect":true,"strictFileInteractability":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify","webauthn:extension:credBlob":true,"webauthn:extension:largeBlob":true,"webauthn:virtualAuthenticators":true},"sessionId":"fe5d5928198e2bbe60a48cc701ae007b"}}
```

[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply test/unit/helper/test_helper.py:53]: oh, this is weird

[the remark this answers is not in the document]
Comment (KazuCocoa) [reply test/unit/helper/test_helper.py:53]: Titus confirmed the _Let response’s [body](https://www.w3.org/TR/2018/REC-webdriver1-20180605/#dfn-body) be a JSON [Object](https://www.w3.org/TR/2018/REC-webdriver1-20180605/#dfn-object) with a key "value" set to the [JSON serialization](https://www.w3.org/TR/2018/REC-webdriver1-20180605/#dfn-json-serialization) of data._ is always. So this response also needs it.

So, as the current selenium implementation to get result via `value` key is expected behavior.
e.g. https://github.com/SeleniumHQ/selenium/blob/trunk/rb/lib/selenium/webdriver/remote/bridge.rb#L592

[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply test/unit/helper/test_helper.py:53]: then the spec is quite confusing in createSession part of it

**Comment by mykola-mokhnach** [comment]
> Should value in test code revert?
> appium/webdriver/webdriver.py looks good.

I have both cases now in tests

**Review by KazuCocoa — APPROVED** [review]
okay. 
I'd prefer to keep `value` in test_helper.py (not webdriver_test.py) to keep the current appium 2.0 response as mocks, but no issues are there.

**Pull request merged**